### PR TITLE
fix(libsinsp): make sure formatted bytebuf doesn't contain irregular …

### PR DESCRIPTION
**Pull request by mrgian**
…characters

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

/area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix(libsinsp): make sure formatted bytebuf doesn't contain irregular characters
```

### CODE REVIEW

File: userspace/libsinsp/sinsp_filtercheck.cpp [hunk 1]
```diff
@@ -1073,20 +1073,9 @@ char* sinsp_filter_check::rawval_to_string(uint8_t* rawval,
...
-			else
-			{
-				auto copy_len = std::min(len, (uint32_t) STRPROPERTY_STORAGE_SIZE);
-				m_getpropertystr_storage.resize(STRPROPERTY_STORAGE_SIZE);
-				memcpy(m_getpropertystr_storage.data(), rawval, copy_len);
-				m_getpropertystr_storage.data()[copy_len] = 0;
-				return m_getpropertystr_storage.data();
-			}
+			m_getpropertystr_storage.resize(STRPROPERTY_STORAGE_SIZE);
+			binary_buffer_to_string(m_getpropertystr_storage.data(), (const char*)rawval, (uint32_t) STRPROPERTY_STORAGE_SIZE - 1, len, sinsp_evt::PF_NORMAL);
```
Comment (jasondellaluce) [inline userspace/libsinsp/sinsp_filtercheck.cpp:1077]:
```suggestion
			binary_buffer_to_string(m_getpropertystr_storage.data(), (const char*)rawval, (uint32_t) STRPROPERTY_STORAGE_SIZE - 1, len, m_inspector->get_buffer_format());
```
Let's use the currently-configured formatting option in the inspector, consistently with the other filter checks

**Review by jasondellaluce — APPROVED** [review]
/approve

**Review by incertum — APPROVED** [review]
/approve

**Pull request merged**